src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/_params.py

```diff
@@ -3,43 +3,54 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+from knack.arguments import CLIArgumentType
 import azure.cli.command_modules.redis._help  # pylint: disable=unused-import
 
 
 def load_arguments(self, _):
     from azure.mgmt.redis.models import RebootType, RedisKeyType, SkuName
+    from azure.cli.core.util import shell_safe_json_parse
     from azure.cli.command_modules.redis._validators import JsonString, ScheduleEntryList
     from azure.cli.core.commands.parameters import get_enum_type  # TODO: Move this into Knack
-    from azure.cli.core.commands.parameters import get_resource_name_completion_list, name_type, tags_type
+    from azure.cli.core.commands.parameters import get_resource_name_completion_list
 
     with self.argument_context('redis') as c:
-        c.argument('name', options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
-                   completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        cache_name = CLIArgumentType(options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
```

Comment:
Something i noticed is that you registered all these arguments at the 'redis' scope. For maintainability, it would be better to register the appropriate arguments at the correct level of specificity.

So if only a command uses a parameter, register it using that scope. For example, 
```
with self.argument_context('redis firewall-rules') as c:
         c.argument('rule_name', help='Name of the firewall rule')
```

Here rule_name is only registered for this limited scope.

See vm/_params.py or even cdn/_params.py (smaller file) for examples of this.

Reply:
Can i ask why though? The parameters  listed here are unique to the whole redis scope even though they are used only within specific commands. I don't see an advantage to keeping them under the scope of the specific command set vs all up, when the parameter is unique

Reply:
It is helpful for maintainability. 
- As it allows multiple commands can share the exact same parameter destination. e.g. name, format, sku etc but have different help or validation. This Allows you to have simpler parameter names, so for example if format and file_format are just `format` under the right scope, your custom methods can have `format` has the file format parameter.
- It makes it clear which commands the parameter belongs to. So for example it is not clear from looking at the code (bug fix, update, etc.) that rule_name belongs to redis firewall-rules or that format belongs to redis export. Etc. 
- appropriately scoped, you would just have format as the parameter to redis export and import. Instead of file_format.

See [vm/_params.py](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py) for how this could be helpful.

I'll leave this up to you, especially seeing the module is rather small.

Reply:
If they are all unique within the `redis` scope then this is fine.